SECEdgar/extractor/EDGARExtractor.py

```diff
@@ -0,0 +1,227 @@
+import re
+import json
+import os.path
+import logging
+import uu
+
+
+class EDGARExtractor():
+
+    # Compile regular expressions
+    # Note that re.search() does not have a starting position argument
+    # but the search() method of a Pattern object
+    # (a compiled regex) has a pos argument
+    re_doc = re.compile("<DOCUMENT>(.*?)</DOCUMENT>", flags=re.DOTALL)
```

Comment:
I feel like the whole regex operation should go in a separate class or a method and should be passed as a dependency.

```FilingExtractor(regex=regex)```

Reply:
I am not sure why we need to pass the regex to the constructor? Aren't the regex used always the same?

Reply:
Not the regex, I was asking whole `regex` operations should be in a separate class. You can define smaller methods.  For Example

``` 
class RegexOperation(object):
     def __init__(string):
           self.string = string

     def extract_comma():
          string_without_comma = re.sub(',', '', self.string)

```

Now pass this class to filing extractor and use the operations.